astropy/io/fits/file.py

```diff
@@ -400,6 +404,36 @@ def _overwrite_existing(self, overwrite, fileobj, closed):
             else:
                 raise IOError("File {!r} already exists.".format(self.name))
 
+    def _test_for_compression(self, obj_or_name, magic, mode, ext=None):
```

Comment:
I changed it anyway since it clarifies the intent. Maybe you were thinking about how if you actually need to test if an object is `None` you need to use `is` rather than `==`?